### ISSUE TITLE
Pasting UTF-8 content anywhere causes crash

### ISSUE THREAD

**Issue by 5HT2** [anegostudios/VintageStory-Issues#1802]
**Game Version:** 1.16.4
**Platform:** Windows / Mac / **Linux** (which?)   
**Modded:** No / **Yes** (which?)  
**SP/MP:** **Singleplayer** / Multiplayer (which?)  

I assume this will happen with other input windows as well.

### Logs
```
Running on 64 bit Linux with 15 GB. 
Game Version: v1.16.4 (Stable)
Loaded Mods: game@1.16.4, egocaribautomapmarkers@2.3.0, carrycapacity@0.6.5, hudclock@2.7.0, prospectorinfo@3.1.2, creative@1.16.4, survival@1.16.4
3/12/2022 12:17:22 PM: Critical error occurred
System.ExecutionEngineException: String conversion error: Partial byte sequence encountered in the input.
  at (wrapper managed-to-native) System.Runtime.InteropServices.Marshal.PtrToStringAnsi(intptr)
  at System.Windows.Forms.XplatUIX11.TranslatePropertyToClipboard (System.IntPtr property) [0x000f2] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.XplatUIX11.UpdateMessageQueue (System.Windows.Forms.XEventQueue queue, System.Boolean allowIdle) [0x00918] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.XplatUIX11.ClipboardRetrieve (System.IntPtr handle, System.Int32 type, System.Windows.Forms.XplatUI+ClipboardToObject converter) [0x0002f] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.XplatUI.ClipboardRetrieve (System.IntPtr handle, System.Int32 type, System.Windows.Forms.XplatUI+ClipboardToObject converter) [0x00000] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.Clipboard.GetDataObject (System.Boolean primary_selection) [0x00039] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.Clipboard.GetDataObject () [0x00000] in <fef229f3059c47f39598bb88ddd21818>:0 
  at System.Windows.Forms.Clipboard.GetText (System.Windows.Forms.TextDataFormat format) [0x0002d] in <fef229f3059c47f39598bb88ddd21818>:0 
  at VSPlatform.XPlatImplLinux.GetClipboardText () [0x00000] in <1b229c8559c64ae8ad26f6ab0a7772cd>:0 
  at Vintagestory.API.Client.GuiElementEditableTextBase.OnKeyDown (Vintagestory.API.Client.ICoreClientAPI api, Vintagestory.API.Client.KeyEvent args) [0x0016a] in <c35009a825774beb829bbd8af877cce6>:0 
  at Vintagestory.API.Client.GuiElementTextInput.OnKeyDown (Vintagestory.API.Client.ICoreClientAPI api, Vintagestory.API.Client.KeyEvent args) [0x0002d] in <c35009a825774beb829bbd8af877cce6>:0 
  at Vintagestory.API.Client.GuiComposer.OnKeyDown (Vintagestory.API.Client.KeyEvent args, System.Boolean haveFocus) [0x0001a] in <c35009a825774beb829bbd8af877cce6>:0 
  at Vintagestory.API.Client.GuiDialog.OnKeyDown (Vintagestory.API.Client.KeyEvent args) [0x00019] in <c35009a825774beb829bbd8af877cce6>:0 
  at Vintagestory.Client.NoObf.GuiManager.OnKeyDown (Vintagestory.API.Client.KeyEvent args) [0x0002f] in <6b6988c5e6264e7e8b7a32e875df9528>:0 
  at Vintagestory.Client.NoObf.ClientMain.OnKeyDown (Vintagestory.API.Client.KeyEvent args) [0x00061] in <6b6988c5e6264e7e8b7a32e875df9528>:0 
  at _4Bi1AGVssrIO0El8b8ygdS9FIfm._FjiApnx6mBMZGzS3QLMXxETjDmf (Vintagestory.API.Client.KeyEvent ) [0x00000] in <6b6988c5e6264e7e8b7a32e875df9528>:0 
  at _lAuHsYOBOAjBMdATFXwLUtH7N00._FjiApnx6mBMZGzS3QLMXxETjDmf (Vintagestory.API.Client.KeyEvent ) [0x0006e] in <6b6988c5e6264e7e8b7a32e875df9528>:0 
  at Vintagestory.Client.NoObf.ClientPlatformWindows.game_KeyDown (System.Object sender, OpenTK.Input.KeyboardKeyEventArgs e) [0x00082] in <6b6988c5e6264e7e8b7a32e875df9528>:0 
  at (wrapper delegate-invoke) System.EventHandler`1[OpenTK.Input.KeyboardKeyEventArgs].invoke_void_object_TEventArgs(object,OpenTK.Input.KeyboardKeyEventArgs)
  at OpenTK.NativeWindow.OnKeyDown (OpenTK.Input.KeyboardKeyEventArgs e) [0x00000] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.NativeWindow.OnKeyDownInternal (System.Object sender, OpenTK.Input.KeyboardKeyEventArgs e) [0x00000] in <a4886e33f9fc416195d24950e7205b94>:0 
  at (wrapper delegate-invoke) System.EventHandler`1[OpenTK.Input.KeyboardKeyEventArgs].invoke_void_object_TEventArgs(object,OpenTK.Input.KeyboardKeyEventArgs)
  at OpenTK.Platform.NativeWindowBase.OnKeyDown (OpenTK.Input.Key key, System.Boolean repeat) [0x0002e] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.Platform.X11.X11GLNative.ProcessEvents () [0x00713] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.NativeWindow.ProcessEvents (System.Boolean retainEvents) [0x00035] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.NativeWindow.ProcessEvents () [0x00000] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.GameWindow.Run (System.Double updates_per_second, System.Double frames_per_second) [0x000da] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.GameWindow.Run () [0x00000] in <a4886e33f9fc416195d24950e7205b94>:0 
  at _TRhArnJABwZoV9Vt7Ld2l65Xzgg._mhlr8s7wtMvxaJGpVy2Ag90jUBK (_3HJA5NvAIx31LXjscc3HGjXaR6q , System.String[] ) [0x003dc] in <29d88641cda44fa481e9734466805d45>:0 
  at _TRhArnJABwZoV9Vt7Ld2l65Xzgg+_fSeTszd4whwFiRbaTa0d2CyPdfJ._SvbPzpxmSlo5pUMfGCrGwu2xebc () [0x00000] in <29d88641cda44fa481e9734466805d45>:0 
  at _q0zAVOrWbMX04rMA9G3HIFVuGjA._mhlr8s7wtMvxaJGpVy2Ag90jUBK (System.Threading.ThreadStart ) [0x00007] in <6b6988c5e6264e7e8b7a32e875df9528>:0 
-------------------------------
```

**Comment by 5HT2** [comment]
> I assume this will happen with other input windows as well.

Confirmed.

**Comment by Th3Dilli** [comment]
this should be fixed at least since 1.18.8

tested pasting UTF-8 into the world creation world name field and seems fine
